docs/bsp.md

```diff
@@ -726,8 +733,19 @@ trait SourceItem {
     * intended to be manually edited by the user. */
   def generated: Boolean
 }
+
+object SourceItemKind {
+  /** The source item references a normal file.  */
+  val File: Int = 1
+  /** The source item references a directory. */
+  val Directory: Int = 2
+  /** The source item is a generated file or directory. */
+  val Generated: Int = 3
```

Comment:
Can we have `GeneratedFile` and `GeneratedDir` (and, in bsp4j, `GENERATED_FILE` and `GENERATED_DIR`)?

Reply:
I'd say either we have a simple value field with the items 'File`, `Directory`, `GeneratedFile`, `GeneratedDirectory`, or a list of "tags" like this with onle 'File`, `Directory`, `Generated`. I prefer the list because being generated is orthogonal to being a file or directory.

Reply:
Do you have any opinion @olafurpg

Reply:
Sorry for the slow response! I agree the "generated" attribute is orthogonal to the item being a file or directory. I still prefer enumerating all possible valid values (file, dir, generated file, generated dir) instead of allowing invalid representations like `[File, Directory]`. One alternative is to keep `kind: FILE | DIRECTORY` and `generated: Boolean`. I don't have a strong opinion either way.

Reply:
Thinking more about it, I'd favor the `kind: FILE | DIRECTORY` and `generated: Boolean`. The users of this API would more likely be interested in picking the type of the source more than in that being generated or not, so with this approach they only need to compare against `FILE` or `DIRECTORY` instead of doing two comparisons (e.g. `FILE && GENERATED_FILE`)

Reply:
Agree, I went with this and pushed. Please have another quick look.